projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -131,6 +137,34 @@ describe('DatagridComponent', () => {
                     'Expected a different new class to display for the second row.'
                 );
             });
+
+            it('has multi selection capabilities when set to multi', function(this: HasFinderAndGrid): void {
+                this.finder.hostComponent.selectionType = GridSelectionType.Multi;
+                let rowsEmitted = 0;
+                this.finder.hostComponent.selectionChanged = (selection: MockRecord[]) =>
+                    (rowsEmitted = selection.length);
+                this.finder.detectChanges();
+                expect(this.clrGridWidget.getSelectionType()).toBe(GridSelectionType.Multi);
+                this.clrGridWidget.selectRow(1);
+                expect(rowsEmitted).toBe(1);
+            });
+
+            it('has single selection capabilities when set to single', function(this: HasFinderAndGrid): void {
+                this.finder.hostComponent.selectionType = GridSelectionType.Single;
+                let rowsEmitted = 0;
+                this.finder.hostComponent.selectionChanged = (selection: MockRecord[]) =>
```

Comment:
The more idiomatic way is to spy on the function and test what is was called with, then you won't need the variable outside